### ISSUE TITLE
Unable to access content of a previous node from anything other than the immediate next node

### ISSUE THREAD

**Issue by neverether** [n8n-io/n8n#7623]
**Describe the bug**
When trying to access a previous node in an expression, only the node before it will return any data. Trying to access data from a node 2, 3 ,4 etc steps backward will not yeild any data.

**To Reproduce**

1. Create a node `A` that outputs data i.e: Set
2. Create some other node `B` that does literally anything else
3. Create 3rd node `C` that tries to access data from node `A`
4. No data 🐛 
5. Connect node `A` to node `C` directly
6. Data 

**Expected behavior**
Expect data to be accessible from following nodes.

**Environment (please complete the following information):**

- OS: Ubuntu 20
- n8n Version 1.4.1
- Node.js Version 18
- Database system SQLite
- Operation mode main?

**Additional context**
Attached video demonstrating the issue.

https://github.com/n8n-io/n8n/assets/1182105/d9dc1ab1-8833-40cd-984c-e3789ad6e11b

**Comment by Joffcom** [comment]
Hey @neverether,

That is an interesting one, it looks like the data is available but not until the node has input data. I have created `N8N-7035` as our internal ticket to have a look at this.

**Comment by sagararyal** [comment]
I have a similar issue where data from a previous node isn't accessible. In my case, I'm processing an array of records in a batch:
<img width="1426" alt="Screenshot 2023-11-09 at 12 31 37 AM" src="https://github.com/n8n-io/n8n/assets/5401575/6eeafe83-d2c5-4549-9e10-4935bd98136a">


Issue present after updating from 1.10.1 to v1.11.2 / Also present on v1.15.2
DB System: Postgres
on Heroku using queue mode

**Comment by Joffcom** [comment]
Hey @sagararyal,

Your issue is very different and is likely to be related to paired items or the way the workflow has been created. Could you create a post on our support forum and include your workflow so we investigate further.

**Comment by Joffcom** [comment]
We’re taking some time to review and tidy up our issue tracker. It looks like this issue hasn’t had any recent activity for the past 6 months. To keep our backlog focused on current priorities, we’re closing this issue for now.

That said, this doesn’t mean the issue is forgotten! If this is still relevant or you have more information to add, feel free to reopen it or create a new issue and we’d be happy to revisit it.

Thank you for understanding and for contributing to the project!
